tests/parametric/test_otel_span_methods.py

```diff
@@ -348,3 +377,437 @@ def test_otel_get_span_context(self, test_agent, test_library):
+            span_kind=span_kind,
+            attributes=None,
+            test_library=test_library,
+            test_agent=test_agent,
+        )
+
+    @missing_feature(context.library == "golang", reason="Not implemented")
+    @missing_feature(context.library == "java", reason="Not implemented")
+    @missing_feature(context.library == "nodejs", reason="Not implemented")
+    @missing_feature(context.library == "dotnet", reason="Not implemented")
+    @missing_feature(context.library == "python", reason="Not implemented")
+    @missing_feature(context.library == "python_http", reason="Not implemented")
+    def test_otel_span_operation_name_unknown(self, test_agent, test_library):
+        """
+            Tests that the operation name will be set to "otel_unknown" when:
+            - span.kind is not present
+            - no other known attributes for setting the operation name
+        """
+        run_operation_name_test(
+            name="otel_span_name",
+            expected_name="otel_unknown",
+            span_kind=None,
+            attributes=None,
+            test_library=test_library,
+            test_agent=test_agent,
+        )
+
+    @missing_feature(context.library == "golang", reason="Not implemented")
+    @missing_feature(context.library == "java", reason="Not implemented")
+    @missing_feature(context.library == "nodejs", reason="Not implemented")
+    @missing_feature(context.library == "dotnet", reason="Not implemented")
+    @missing_feature(context.library == "python", reason="Not implemented")
+    @missing_feature(context.library == "python_http", reason="Not implemented")
+    def test_otel_span_special_tag_overrides(self, test_agent, test_library):
+        """
+            Tests that the special tags will override expected values
+        """
+        with test_library:
+            with test_library.otel_start_span("otel_span_name", span_kind=SK_SERVER) as span:
+                span.set_attributes({"http.request.method": "GET"})
+                span.set_attributes({"resource.name": "new.name"})
+                span.set_attributes({"operation.name": "Overriden.name"})
+                span.set_attributes({"service.name": "new.service.name"})
+                span.set_attributes({"span.type": "new.span.type"})
+                span.set_attributes({"analytics.event": "true"})
+                span.end_span()
+        traces = test_agent.wait_for_num_traces(1)
+        trace = find_trace_by_root(traces, otel_span(name="new.name"))
+        assert len(trace) == 1
+
+        span = get_span(test_agent)
+        assert span["name"] == "overriden.name"
+        assert span["resource"] == "new.name"
+        assert span["service"] == "new.service.name"
+        assert span["type"] == "new.span.type"
+        assert span["metrics"].get("_dd1.sr.eausr") == "1.0"
+
+
+def run_operation_name_test(name: str, expected_name: str, span_kind: int, attributes: dict, test_library, test_agent):
```

Comment:
Let's change `name` to `resource`. I realize that in the context of OTel, "name" is equivalent to "resource name", but I fear this will be confusing when `expected_name` is also there and means something different. So I'd suggest: `expected_operation_name`, `resource`, `span_kind`, `attributes`